xcube/webapi/ows/stac/controllers.py

```diff
@@ -385,6 +386,51 @@ def get_collection_queryables(
     return schema.to_dict()
 
 
+def get_collection_schema(
+    ctx: DatasetsContext, base_url: str, collection_id: str
+) -> dict:
+    """Return a JSON schema for a dataset's data variables
+
+    See links in
+    https://docs.ogc.org/DRAFTS/19-087.html#_collection_schema_response_collectionscollectionidschema
+    for links to a metaschema defining the schema.
+
+    :param ctx: a datasets context
+    :param base_url: the base URL at which this API is being served
+    :param collection_id: the ID of a dataset in the provided context
+    :return: a JSON schema representing the specified dataset's data variables
+    """
+    if collection_id == DEFAULT_COLLECTION_ID:
+        # The default collection contains multiple datasets, so a range
+        # schema doesn't make sense.
+        raise ValueError(f'Invalid collection ID {DEFAULT_COLLECTION_ID}')
+    _assert_valid_collection(ctx, collection_id)
+
+    ml_dataset = ctx.get_ml_dataset(collection_id)
+    ds = ml_dataset.base_dataset
+
+    def get_title(var_name: str) -> str:
+        attrs = ds[var_name].attrs
+        return attrs['long_name'] if 'long_name' in attrs else var_name
+
+    return {
+        '$schema': 'https://json-schema.org/draft/2020-12/schema',
```

Comment:
Maybe put this string into a constant.

Reply:
I don't see much need, but don't really have anything against it either, so done.